### PR TITLE
Miscellaneous Website Documentation

### PR DESCRIPTION
**Pull request by mabruzzo**
This updates some coding guidelines. 

It also updates the advice for getting help on the getting started page (Somebody mentioned to me that they had emailed the other address and their email seems to have fallen through the cracks. It seems much more likely for people to get a prompt response if they contact the mailing list -- since there are a lot more eyes on the mailing list at a given time).

### CODE REVIEW

**Review by gregbryan — APPROVED** [review]

File: doc/source/devel/coding-guidelines.rst [hunk 1]
```diff
@@ -16,48 +16,88 @@ a working document and always open to suggestions for changes.
...
+
 
 ==============
 Naming classes
 ==============
 
-Enzo-E classes all begin with ``Enzo``, followed by the (capitalized)
+Enzo-E classes all begin with ``Enzo``.
+
+Subclasses of the Cello-Hierarcy
```
Comment (jwise77) [inline doc/source/devel/coding-guidelines.rst:82]:
```suggestion
Subclasses of the Cello-Hierarchy
```

File: doc/source/devel/coding-guidelines.rst [hunk 2]
```diff
@@ -68,6 +108,58 @@ the files ``enzo_``\ *class-name*\ ``.hpp`` and ``enzo_``\ *class-name*\ ``.cpp`
 | Restriction         | ``EnzoRestrict``\ *Name*  | ``Restrict``     |
 +---------------------+---------------------------+------------------+
 
+Classes outside of the Cello-Hierarcy
```
Comment (jwise77) [inline doc/source/devel/coding-guidelines.rst:111]:
```suggestion
Classes outside of the Cello-Hierarchy
```

File: doc/source/devel/coding-guidelines.rst [hunk 3]
```diff
@@ -95,6 +187,95 @@ residing on a different processing element in a different memory
...
+     The Google C++ Style Guide calls this "Interface Inheritance".
+
+   * Inheritance can also be used in other cases as a mechansim for
+     reusing code.
+
+       * For example, one might want to make slight changes to the
+         behavior of a base class by subclassing it.
+       * Unfortunately, this practice easily/commonly produces code
+         where the control flow is less explicit, which makes the code
+         harder to reason about.If the base class has a chain of
```
Comment (jwise77) [inline doc/source/devel/coding-guidelines.rst:213]:
```suggestion
         harder to reason about. If the base class has a chain of
```

**Review by jwise77 — APPROVED** [review]
This is a great addition to the docs, @mabruzzo. Thanks very much! I caught a few typos, which I've suggested their fixes. We can merge this after you accept these suggestions.

**Comment by jwise77** [comment]
> This is a great addition to the docs, @mabruzzo. Thanks very much! I caught a few typos, which I've suggested their fixes. We can merge this after you accept these suggestions.

I forgot that I had the ability to make these changes. I've corrected these typos, and now we can merge!

**Pull request merged**